src/tunacode/types/__init__.py

```
@@ -104,3 +79,107 @@
+        """Accumulate cost from another usage cost object."""
+        self.input += other.input
+        self.output += other.output
+        self.cache_read += other.cache_read
+        self.cache_write += other.cache_write
+        self.total += other.total
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> "UsageCost":
+        """Build cost object from a usage payload."""
+        if not isinstance(data, dict):
+            raise ValueError("usage.cost must be a dict")
```

Ruff (0.15.11)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)